From 751832af6dfa8b2b6b5fcdf75c94f83c64739f89 Mon Sep 17 00:00:00 2001
From: Sebastian Ramacher <sebastian@ramacher.at>
Date: Sat, 18 Jul 2026 21:09:54 +0200
Subject: [PATCH] Allow shell completions to be disabled

This option is introduced to disable them during flathub builds.
---
 data/meson.build  | 66 ++++++++++++++++++++++++-----------------------
 meson_options.txt |  7 ++++-
 2 files changed, 40 insertions(+), 33 deletions(-)

diff --git a/data/meson.build b/data/meson.build
index fde12f61..acfbd909 100644
--- a/data/meson.build
+++ b/data/meson.build
@@ -53,38 +53,40 @@ if appstreamcli.found()
   )
 endif
 
-conf_data = configuration_data()
-conf_data.set('PLUGINDIR', join_paths(prefix, plugindir))
-bash_completion = configure_file(
-  input: 'bash-completion.in',
-  output: 'zathura',
-  configuration: conf_data
-)
-zsh_completion = configure_file(
-  input: 'zsh-completion.in',
-  output: '_zathura',
-  configuration: conf_data
-)
-fish_completion = configure_file(
-  input: 'fish-completion.in',
-  output: 'zathura.fish',
-  configuration: conf_data
-)
+if get_option('shell-completions').allowed()
+  conf_data = configuration_data()
+  conf_data.set('PLUGINDIR', join_paths(prefix, plugindir))
+  bash_completion = configure_file(
+    input: 'bash-completion.in',
+    output: 'zathura',
+    configuration: conf_data
+  )
+  zsh_completion = configure_file(
+    input: 'zsh-completion.in',
+    output: '_zathura',
+    configuration: conf_data
+  )
+  fish_completion = configure_file(
+    input: 'fish-completion.in',
+    output: 'zathura.fish',
+    configuration: conf_data
+  )
 
-bash_comp = dependency('bash-completion', required: false)
-if bash_comp.found()
-  bash_compdir = bash_comp.get_variable(pkgconfig: 'completionsdir')
-else
-  bash_compdir = join_paths(datadir, 'bash-completion', 'completions')
-endif
+  bash_comp = dependency('bash-completion', required: false)
+  if bash_comp.found()
+    bash_compdir = bash_comp.get_variable(pkgconfig: 'completionsdir')
+  else
+    bash_compdir = join_paths(datadir, 'bash-completion', 'completions')
+  endif
 
-fish_comp = dependency('fish', required: false)
-if fish_comp.found()
-  fish_compdir = fish_comp.get_variable(pkgconfig: 'completionsdir')
-else
-  fish_compdir = join_paths(datadir, 'fish', 'vendor_completions.d')
-endif
+  fish_comp = dependency('fish', required: false)
+  if fish_comp.found()
+    fish_compdir = fish_comp.get_variable(pkgconfig: 'completionsdir')
+  else
+    fish_compdir = join_paths(datadir, 'fish', 'vendor_completions.d')
+  endif
 
-install_data(bash_completion, install_dir: bash_compdir)
-install_data(zsh_completion, install_dir: join_paths(datadir, 'zsh', 'site-functions'))
-install_data(fish_completion, install_dir: fish_compdir)
+  install_data(bash_completion, install_dir: bash_compdir)
+  install_data(zsh_completion, install_dir: join_paths(datadir, 'zsh', 'site-functions'))
+  install_data(fish_completion, install_dir: fish_compdir)
+endif
diff --git a/meson_options.txt b/meson_options.txt
index fb4cd296..817b24de 100644
--- a/meson_options.txt
+++ b/meson_options.txt
@@ -18,6 +18,11 @@ option('manpages',
   value: 'auto',
   description: 'manual pages'
 )
+option('shell-completions',
+  type: 'feature',
+  value: 'auto',
+  description: 'install bash/fish/zsh completion files'
+)
 option('tests-x11',
   type: 'feature',
   value: 'auto',
@@ -32,4 +37,4 @@ option('convert-icon',
   type: 'feature',
   value: 'auto',
   description: 'convert icon to PNG'
-)
\ No newline at end of file
+)
-- 
2.53.0

